real-main/app/handlers/appsync/handlers.py

```diff
@@ -401,6 +406,19 @@ def set_user_apns_token(caller_user, arguments, **kwargs):
     return caller_user.serialize(caller_user.id)
 
 
+@routes.register('Mutation.setUserDatingStatus')
+@validate_caller
+@update_last_client
+def set_user_dating_status(caller_user, arguments, **kwargs):
+    status = arguments.get('status')
```

Comment:
```suggestion
    status = arguments['status']
```

fail early and often. `status` is required, so better if this fails on access than it pulls out `None` and then fails further down the code.